File: app/src/test/java/com/kieran/winnipegbustests/AppTests.java
package com.kieran.winnipegbustests;

import com.kieran.winnipegbusbackend.LoadResult;
import com.kieran.winnipegbusbackend.RouteKey;
import com.kieran.winnipegbusbackend.StopSchedule;
import com.kieran.winnipegbusbackend.StopTime;
import com.kieran.winnipegbusbackend.TransitApiManager;

import junit.framework.Assert;

import org.json.JSONObject;
import org.junit.Test;

public class AppTests {
    @Test
    public void DateParseMilliseconds() {
        StopTime time = StopTime.convertStringToStopTime("2016-07-10T12:07:30");
        Assert.assertEquals(time.getMilliseconds(), 1468170450000L);
    }

    @Test
    public void RouteNumberSearchQueryGenerationShouldBeValid() {
        String url = TransitApiManager.generateSearchQuery(36).getQueryUrl();

        Assert.assertNotNull(TransitApiManager.getJson(url));
    }

    @Test
    public void RouteKeySearchQueryGenerationShouldBeValid() {
        RouteKey key = new RouteKey(36, 1, 'M');
        String url = TransitApiManager.generateSearchQuery(key).getQueryUrl();

        Assert.assertNotNull(TransitApiManager.getJson(url));
    }

    @Test
    public void ServiceAdvisoriesUrlGenerationShouldBeValid() {
        String url = TransitApiManager.generateServiceAdvisoriesUrl();

        Assert.assertNotNull(TransitApiManager.getJson(url));
    }


    @Test
    public void GenericSearchQueryGenerationShouldBeValid() {
        String url = TransitApiManager.generateSearchQuery("Grant").getQueryUrl();

        Assert.assertNotNull(TransitApiManager.getJson(url));
    }

    @Test
    public void StopFeaturesUrlGenerationShouldBeValid() {
        String url = TransitApiManager.generateStopFeaturesUrl(10545);

        Assert.assertNotNull(TransitApiManager.getJson(url));
    }


    @Test
    public void StopNumberUrlGenerationShouldBeValid() {
        String url = TransitApiManager.generateStopNumberURL(10545, 11, new StopTime(), null);

        Assert.assertNotNull(TransitApiManager.getJson(url));
    }

    @Test
    public void StopScheduleCreationShouldReturnScheduledStops() {
        String url = TransitApiManager.generateStopNumberURL(10545, 11, new StopTime(), null);
        LoadResult<JSONObject> result = TransitApiManager.getJson(url);
        JSONObject dom = result.getResult();

        StopSchedule stopSchedule = new StopSchedule(dom);

        Assert.assertTrue(stopSchedule.getScheduledStopsSorted().size() > 0);
    }

    @Test
    public void StopTimeShouldCreateCorrect12HourString() {
        StopTime time = new StopTime(1468606042000L);

        Assert.assertEquals(time.to12hrTimeString(), "1:07p");
    }

    @Test
    public void StopTimeShouldCreateCorrect24HourString() {
        StopTime time = new StopTime(1468606042000L);

        Assert.assertEquals(time.to24hrTimeString(), "13:07");
    }
}

File: app/src/test/java/com/kieran/winnipegbustests/AppTests.kt
package com.kieran.winnipegbustests

import com.kieran.winnipegbusbackend.winnipegtransit.WinnipegTransitTripIdentifier
import com.kieran.winnipegbusbackend.StopSchedule
import com.kieran.winnipegbusbackend.StopTime
import com.kieran.winnipegbusbackend.winnipegtransit.TransitApiManager

import junit.framework.Assert

import org.junit.Test

class AppTests {
    @Test
    fun DateParseMilliseconds() {
        val time = StopTime.convertStringToStopTime("2016-07-10T12:07:30")
        Assert.assertEquals(time!!.milliseconds, 1468170450000L)
    }

    @Test
    fun RouteNumberSearchQueryGenerationShouldBeValid() {
        val url = TransitApiManager.generateSearchQuery(36).queryUrl

        Assert.assertNotNull(TransitApiManager.getJson(url))
    }

    @Test
    fun RouteKeySearchQueryGenerationShouldBeValid() {
        val key = WinnipegTransitTripIdentifier(36, 1, 'M')
        val url = TransitApiManager.generateSearchQuery(key).queryUrl

        Assert.assertNotNull(TransitApiManager.getJson(url))
    }

    @Test
    fun ServiceAdvisoriesUrlGenerationShouldBeValid() {
        val url = TransitApiManager.generateServiceAdvisoriesUrl()

        Assert.assertNotNull(TransitApiManager.getJson(url))
    }


    @Test
    fun GenericSearchQueryGenerationShouldBeValid() {
        val url = TransitApiManager.generateSearchQuery("Grant").queryUrl

        Assert.assertNotNull(TransitApiManager.getJson(url))
    }

    @Test
    fun StopFeaturesUrlGenerationShouldBeValid() {
        val url = TransitApiManager.generateStopFeaturesUrl(10545)

        Assert.assertNotNull(TransitApiManager.getJson(url))
    }


    @Test
    fun StopNumberUrlGenerationShouldBeValid() {
        val url = TransitApiManager.generateStopNumberURL(10545, 11, StopTime(), null)

        Assert.assertNotNull(TransitApiManager.getJson(url))
    }

    @Test
    fun StopScheduleCreationShouldReturnScheduledStops() {
        val url = TransitApiManager.generateStopNumberURL(10545, 11, StopTime(), null)
        val result = TransitApiManager.getJson(url)
        val dom = result.result

        val stopSchedule = StopSchedule(dom!!)

        Assert.assertTrue(stopSchedule.scheduledStopsSorted.isNotEmpty())
    }

    @Test
    fun StopTimeShouldCreateCorrect12HourString() {
        val time = StopTime(1468606042000L)

        Assert.assertEquals(time.to12hrTimeString(), "1:07p")
    }

    @Test
    fun StopTimeShouldCreateCorrect24HourString() {
        val time = StopTime(1468606042000L)

        Assert.assertEquals(time.to24hrTimeString(), "13:07")
    }
}
